### PR TITLE
[Home Page Picker] Adds polling mechanism for not generated thumbnails 

### PR DESCRIPTION
**Pull request by antonis**
**Supplemental fix for:** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2718

**Depends on:**
- https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1771

## Description

This PR implements a custom Glide loader that prevents following redirects and applies a retry policy with a specific number of retries (10).

This is needed because the mshot backend service redirects to a loading gif image when the thumbnail is not ready. This occurs when the thumbnail has not been accessed recently for the specific language and viewport size. When this thumbnail is requested it is cached on the server and is readily available according to the server cache policy.

## To test:

<details>
<summary><strong>To disable or enable the development version of Home Page Picker</strong></summary>

- Open the app from the build that allows FeatureFlags such as a PR build or a local development build
    - By default, the modal layout picker is enabled only in development builds.
    - From the site page:
        1. Click on your Gravatar.
        1. Click on App Settings.
        1. Click on Test feature configuration.
        1. Toggle "HomePagePickerFeatureConfig" to enable or disable the picker
        1. Restart the App
</details>

<details>
<summary><strong>To start the Home Page Picker screen</strong></summary>

1. Start the site creation flow (e.g. *Choose site > Add button*)
1. Select the *Create WordPress.com site* option
</details>

1. Reinstall the app or clean the app cache (Settings > Apps > WordPress > Storage & cache > Clear)
1. Login
1. Set the application language to one that has no cached thumbnails on server (try `ja` or `ko`)
1. Start the Home Page Picker 
1. Expect to see the ghost images for a short time then see the images load

*Note: an alternative way to test this is to tweak the thumbnail url to request a different viewport [programmatically](https://github.com/wordpress-mobile/WordPress-Android/pull/13386/commits/d29437dd40ce928262465e9daa33b92c9c15bfcf)*

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/modules/WordPressGlideModule.kt [hunk 1]
```diff
@@ -22,6 +24,7 @@ import javax.inject.Named
 @GlideModule
 class WordPressGlideModule : AppGlideModule() {
     @Inject @field:Named("custom-ssl") lateinit var requestQueue: RequestQueue
+    @Inject @field:Named("noRedirects") lateinit var noRedirectsRequestQueue: RequestQueue
```
Comment (mkevins) [inline WordPress/src/main/java/org/wordpress/android/modules/WordPressGlideModule.kt]: NP: I think we can remove `field` here:

```suggestion
    @Inject @Named("custom-ssl") lateinit var requestQueue: RequestQueue
    @Inject @Named("noRedirects") lateinit var noRedirectsRequestQueue: RequestQueue
```

**Review by mkevins — COMMENTED** [review]
The code looks good, and I've tested via Pixel 3a (physical device) and I believe it is working as expected. I noticed that the placeholder image was that of a "cube", so I just wanted to confirm that this is expected (i.e. a cube image is eventually replaced by the screenshot thumbnails). Does that sound right?

**Comment by antonis** [comment]
Hello @mkevins 👋 

Thank you for taking the time to review this 🙇 

> I noticed that the placeholder image was that of a "cube", so I just wanted to confirm that this is expected (i.e. a cube image is eventually replaced by the screenshot thumbnails). Does that sound right?

Thank you for bringing this out 👍 
I'm [using](https://github.com/wordpress-mobile/WordPress-Android/pull/13386/files#diff-9e9bb051dea794d926d61427f20715cf36bda5fccf389437d477aa5cce005333R257) the pre-existing `ImageType.THEME` as a placeholder. This is intentional (and did the same with the Page Layout Picker). This is not in the design specs but it seemed to fit the purpose at the time.

I'm adding a screenshot here for reference and in case @iamthomasbishop has feedback on this.
<image width="380" src="https://user-images.githubusercontent.com/304044/99217283-5eb4d900-27e0-11eb-9bd5-dddb1a20110a.png"/>

Note: Most of the time this placeholder does not appear at all (when the images are cached) or appear very shortly (when the images are already generated on the server).

**Review by mkevins — APPROVED** [review]
> This is intentional

Thanks for confirming :+1: In that case, LGTM! Nice work Antonis!

**Review by chipsnyder — APPROVED** [review]
LGTM! Nice work @antonis!

**Pull request merged**